scripts/npm-safe-publish.sh

```diff
@@ -68,36 +66,32 @@ if [[ "$NPM_SHOW" != "$NPM_LS" ]]; then
       exit 1
     fi
 
-    # check if current commit is tagged with the requested version
-    if [[ -z "$(git tag --points-at HEAD "$PACKAGE@$NPM_LS")" ]]; then
-      echo "Current commit is not tagged with the requested version. Aborting."
-      exit 1
-    fi
-
     # check if otp is set
-    if [[ -z "$OTP" ]]; then
+    if [[ -z "$OTP" && -z "$NODE_AUTH_TOKEN" ]]; then
       echo "Must specify the NPM one-time password using the --otp option."
       exit 1
     fi
 
     echo "Publishing $PACKAGE@$NPM_LS to NPM."
-    npm publish --otp=$OTP
+    if [[ -z "$NODE_AUTH_TOKEN" ]]; then
+      npm publish --otp="$OTP"
+    else
+      npm publish --access=public
+    fi
 
+    # shellcheck disable=SC2181
     if [[ "$?" != 0 ]]; then
       echo "Publish failed. Aborting."
       exit 1
     fi
 
-    curl -X POST \
-      -d api_key=$DEVCYCLE_PROD_SLEUTH_API_TOKEN \
-      -d environment=production \
-      -d sha=$SHA https://app.sleuth.io/api/1/deployments/taplytics/js-sdks-2/register_deploy
   else
     echo "NPM Publish Local"
     npm publish --local
   fi
 else
   echo "Versions are the same ($NPM_SHOW = $NPM_LS). Not pushing"
+  exit 0
```

Comment:
I think we want it to go down the Deprecated code path even if the existing versions are the same.

Reply:
If the existing versions are the same - nothing should be published - NPM Publish will throw an exit code if we do - which is why I wanted to terminate early.